src/main/java/com/conveyal/r5/transit/TransportNetworkCache.java

```diff
@@ -108,15 +106,15 @@ public synchronized TransportNetwork getNetwork (String networkId) {
     }
 
     /**
-     * Find or create a TransportNetwork for the given
+     * Find or create a TransportNetwork for the given scenario ID.
      * By design a particular scenario is always defined relative to a single base graph (it's never applied to multiple
      * different base graphs). Therefore we can look up cached scenario networks based solely on their scenarioId
      * rather than a compound key of (networkId, scenarioId).
      *
      * The fact that scenario networks are cached means that PointSet linkages will be automatically reused when
      * the scenario is found by its ID and reused.
      *
-     * TODO LinkedPointSets keep a reference back to a StreetLayer which means that the network will not be completely garbage collected upon network switch
+     * TODO why are we passing in the request instead of the Scenario (which should contain its own ID)?
```

Comment:
I believe because this function handles both scenarios in profile requests as well as scenarios pulled from S3.